src/Seeker/S3ObjectSeeker.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Mingulay\Seeker;
+
+use Exception;
+use InvalidArgumentException;
+use Mingulay\SeekerInterface;
+
+class S3ObjectSeeker implements SeekerInterface {
+
+  /**
+   * The valid s3:// wrapper path to an S3 Object.
+   *
+   * @var string
+   */
+  protected $path;
+
+  /**
+   * The file pointer.
+   *
+   * @var false|resource
+   */
+  protected $stream;
+
+  /**
+   * The size of the file.
+   *
+   * @var int
+   */
+  protected $size;
+
+
+  /**
+   * Create a new S3ObjectSeeker object.
+   *
+   * @param string $path The  streamwrapper prefixed path to an S3 Object.
+   *
+   * @throws InvalidArgumentException if the path is not accessible.
+   */
+  public function __construct(string $path) {
+    $this->path = $path;
+    try {
+      $this->size = filesize($path);
+      if ($this->size === FALSE) {
+        throw new InvalidArgumentException(
+          "The provided S3 Object could not be opened"
+        );
+      }
+    } catch (Exception $e) {
+      throw new InvalidArgumentException(
+        "The provided S3 Object could not be opened: " . $e->getMessage()
+      );
+    }
+  }
+
+  /**
+   * Destructor function to ensure the file descriptor is closed.
+   */
+  public function __destruct() {
+    if ($this->stream) {
```

Comment:
Ah, this is sensible! I should do the same in LocalFileSeeker, which just does a blind `fclose` at the moment.

Reply:
Implemented in LocalFileSeeker now!